WordPress/Classes/System/SplitViewRootPresenter.swift

```diff
@@ -33,6 +33,27 @@ final class SplitViewRootPresenter: RootViewPresenter {
         sidebarViewModel.navigate = { [weak self] in
             self?.navigate(to: $0)
         }
+
+        // Automatically switch to a site or show the sign in screen, when the current blog is removed.
+        NotificationCenter.default
+            .publisher(for: .NSManagedObjectContextObjectsDidChange, object: ContextManager.shared.mainContext)
+            .sink { [weak self] in
```

Comment:
(nit) Extract it to a separate method.

Reply:
Addressed in https://github.com/wordpress-mobile/WordPress-iOS/pull/23531/commits/4bcb32dfb51a2f51c9fc1105be5fae1113c1f60e